nautobot/core/filters.py

```diff
@@ -462,6 +445,21 @@ class SearchFilter(MappedPredicatesFilterMixin, django_filters.CharFilter):
     label = "Search"
 
 
+class TagFilter(NaturalKeyOrPKMultipleChoiceFilter):
+    """
+    Match on one or more assigned tags. If multiple tags are specified (e.g. ?tag=foo&tag=bar), the queryset is filtered
+    to objects matching all tags.
+    """
+
+    def __init__(self, *args, **kwargs):
+        from nautobot.extras import models as extras_models  # avoid circular import
```

Comment:
Maybe? as we not using any other class from this module
```suggestion
        from nautobot.extras.models import Tag  # avoid circular import
```